pkg/exchange/coinbase/stream.go

```diff
@@ -21,10 +33,146 @@ type Stream struct {
 	activeMessageCallbacks   []func(m *ActiveMessage)
 }
 
-func NewStream() *Stream {
+func NewStream(
+	apiKey string,
+	passphrase string,
+	secretKey string,
+) *Stream {
 	s := Stream{
 		StandardStream: types.NewStandardStream(),
+		apiKey:         apiKey,
+		passphrase:     passphrase,
+		secretKey:      secretKey,
 	}
 	s.SetParser(s.parseMessage)
+	s.SetDispatcher(s.dispatchEvent)
+	s.SetEndpointCreator(createEndpoint)
+
+	// public handlers
+	s.OnConnect(s.handleConnect)
+
 	return &s
 }
+
+func (s *Stream) dispatchEvent(e interface{}) {
+	switch e := e.(type) {
+	case *StatusMessage:
+		s.EmitStatusMessage(e)
+	case *AuctionMessage:
+		s.EmitAuctionMessage(e)
+	case *RfqMessage:
+		s.EmitRfqMessage(e)
+	case *TickerMessage:
+		s.EmitTickerMessage(e)
+	case *ReceivedMessage:
+		s.EmitReceivedMessage(e)
+	case *OpenMessage:
+		s.EmitOpenMessage(e)
+	case *DoneMessage:
+		s.EmitDoneMessage(e)
+	case *MatchMessage:
+		s.EmitMatchMessage(e)
+	case *ChangeMessage:
+		s.EmitChangeMessage(e)
+	case *ActiveMessage:
+		s.EmitActiveMessage(e)
+	default:
+		log.Warnf("skip dispatching msg due to unknown message type: %T", e)
+	}
+}
+
+func createEndpoint(ctx context.Context) (string, error) {
+	return wsFeedURL, nil
+}
+
+type channelType struct {
+	Name       string   `json:"name"`
+	ProductIDs []string `json:"product_ids,omitempty"`
+}
+
+type websocketCommand struct {
+	Type       string        `json:"type"`
+	Channels   []channelType `json:"channels"`
+	Signature  *string       `json:"signature,omitempty"`
+	Key        *string       `json:"key,omitempty"`
+	Passphrase *string       `json:"passphrase,omitempty"`
+	Timestamp  *string       `json:"timestamp,omitempty"`
+}
+
+func (s *Stream) handleConnect() {
+	// subscribe to channels
+	if len(s.Subscriptions) == 0 {
+		return
+	}
+
+	subProductsMap := make(map[string][]string)
+	for _, sub := range s.Subscriptions {
+		strChannel := string(sub.Channel)
+		// "rfq_matches" allow empty symbol
+		if sub.Channel != "rfq_matches" && len(sub.Symbol) == 0 {
```

Comment:
define rfq_matches as constant on global

Reply:
What's the meaning of rfa_matchers?

Reply:
`rfq` stands for "Request For Quote".

I will make it a global const.

Reply:
For details on rfq, please refer to the following link:
https://help.coinbase.com/en/exchange/trading-and-funding/request-for-quote-rfq